core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1127,6 +1127,23 @@ object TestUtils extends Logging {
       throw new IllegalStateException(s"Cannot get topic: $topic, partition: $partition in server metadata cache"))
   }
 
+  /**
+   * Wait until the kraft broker metadata have caught up to the controller
+   */
+  def waitForKRaftBrokerMetadataCatchupController(
```

Comment:
I think we can make the name more concise. How about `ensureConsistentKRaftMetadata` or something like that?